proposals/lambda-improvements.md

```diff
@@ -41,26 +41,36 @@ app.MapAction([HttpPost("/")] ([FromBody] Todo todo) => todo);
 ```
 
 ## Attributes
-Attributes may be added to lambda expressions.
+Attributes may be added to lambda expressions and lambda parameters.
+To avoid ambiguity between method attributes and parameter attributes, a lambda expression with attributes must use a parenthesized parameter list.
+Parameter types are not required.
 ```csharp
-f = [MyAttribute] x => x;          // [MyAttribute]lambda
-f = [MyAttribute] (int x) => x;    // [MyAttribute]lambda
-f = [MyAttribute] static x => x;   // [MyAttribute]lambda
-f = [return: MyAttribute] () => 1; // [return: MyAttribute]lambda
+f = [A] () => { };        // [A]lambda
+f = [return:A] x => x;    // syntax error at '=>'
+f = [return:A] (x) => x;  // [A]lambda
+f = [A] static x => x;    // syntax error at '=>'
```

Comment:
What is the correct way to do this? `[A] static (x) = x` or `static [A] (x) = x`? 

In C++ *capture list* always comes first (analogous to *attribute list*) but specifiers comes after params. So that probably wouldn't be useful. See https://en.cppreference.com/w/cpp/language/lambda